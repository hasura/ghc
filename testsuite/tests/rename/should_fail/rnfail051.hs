-- trac #2033: This was previously used to fail when the renamer
-- didn't check for a view
-- /pattern/ being used in an /expression/ context
--
-- But due to GHC proposal #281 all arrows in an expression context
-- are function arrow type now

module RnFail051 where

main :: IO ()
main = wrapper (_ -> putStrLn "_")

wrapper :: (String -> IO ()) -> IO ()
wrapper f = f ""
