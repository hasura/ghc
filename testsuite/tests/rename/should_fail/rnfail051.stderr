rnfail051.hs:11:17: error: [GHC-31244]
    Unexpected type syntax in the expression
    Suggested fix:
      Perhaps you intended to use the ‘RequiredTypeArguments’ extension

